Xkcd-Comics/ViewModels/ComicsViewModel.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  ComicsVM.swift
+//  Xkcd-Comics
+//
+//  Created by Jia-Jiuan Tsai on 2021-11-24.
+//
+
+import Foundation
+
+class ComicsViewModel: ObservableObject {
+    @Published private(set) var comic = Comic(num: 1, title: "Default", img: "Default", alt: "Default", year: "Default", month: "Default")
+    @Published private(set) var comics: [Comic] = []
+    @Published private(set) var isSaved = false
+    @Published var isShowingNextComicAlert = false
+    @Published var isShowingPreviousComicAlert = false
+    // Explanation
+    @Published var isShowingExplanation = false
+    @Published private(set) var isShowingInfo = false
+    // Search
+    @Published private(set) var isShowingSearch = false
+    @Published var searchValue = ""
+    let fetchingService: FetchingService
+    let parsingService: ParsingService
+    
+    init(fetchingService: FetchingService, parsingService: ParsingService) {
+        self.fetchingService = fetchingService
+        self.parsingService = parsingService
+        fetchComic()
+    }
+    
+    func fetchComic() {
+        fetchingService.fetchComic(fetchingURL: .latestComic) { result in
+            switch result {
+            case .success(let comic):
+                DispatchQueue.main.async {
+                    self.comic = comic
+                    self.checkIfSaved(comic: comic)
```

Comment:
Code duplication: 

Every time `self.comic = comic` is called, you also have `self.checkIfSaved(comic: comic)`. It's easy to forget that the two lines always have to go together, and it's going to be time-consuming and error-prone if we need to add more steps or change something for all of them in the future. 

If the code that handles a success response is more than a simple setting of a property, and used more than once, it needs to be centralized. 

My preferred way of doing this is to add a `didSet` to the `comic` property. This way, we ensure that any time `comic` is set, the saved status is checked. 

You could also create a method called something like `private func handleComic(_ comic: Comic)` which calls both lines.